design-principles.md

```diff
@@ -0,0 +1,45 @@
+# Design Principles
+
+There are exceptions to these principles, but these are general guidelines the project strives to adhere to.
+
+## General
+
+- Bootkube should be a single-use tool, which only runs on the first node in a cluster.
+    - An exception is the `recovery` subcommand, discussed below.
+- Bootkube should not be required to run to add new nodes to a cluster.
+    - For example, adding nodes should not require a `bootkube join` command.
```

Comment:
Perhaps add a short sentence on how to do it instead and/or why there is no `join` command?